### PR TITLE
web: stop reporting the `ResizeObserver loop limit exceeded` error to Sentry

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context
 
The `ResizeObserver loop limit` error means that `ResizeObserver` was not able to deliver all observations within a single animation frame. It doesn't break the functionality of the application. The W3C considers converting this error to a warning:   https://github.com/w3c/csswg-drafts/issues/5023. We can safely ignore it in the production environment to avoid hammering Sentry and other libraries relying on `window.addEventListener('error', callback)`.

And we _need_ to ignore it because it eats up our Sentry quota quickly.:
Screenshot 2021-10-05 at 12 03 20

## Changes

- Catch the `ResizeObserver loop limit` error to prevent it from going to the Sentry.
- Move the `ResizeObserver` callback into RAF to prevent errors from happening.
    - Note: this doesn't prevent all `ResizeObserver loop limit` errors. Some third-party libraries use it without RAF; that's why catching the error in production is necessary.

### CODE REVIEW

File: client/web/src/util/dom.ts [hunk 1]
```diff
@@ -8,17 +8,25 @@ import { useObservable } from '@sourcegraph/shared/src/util/useObservable'
...
+            // Move `ResizeObserver` measurements into a RAF to avoid the "ResizeObserver loop limit exceeded" error.
+            // See the thread for background info: https://github.com/WICG/resize-observer/issues/38
+            animationFrameID = window.requestAnimationFrame(() => {
+                observer.next(head(entries))
+            })
         })
         resizeObserver.observe(target)
 
         return function unsubscribe() {
+            window.cancelAnimationFrame(animationFrameID)
```
Comment (eseliger) [inline client/web/src/util/dom.ts:25]: Would it complain if you pass it undefined?

**Review by eseliger — APPROVED** [review]

File: client/web/src/sentry.ts [hunk 2]
```diff
@@ -2,6 +2,23 @@ import * as sentry from '@sentry/browser'
...
+     * the functionality of the application. The W3C considers converting this error to a warning:
+     * https://github.com/w3c/csswg-drafts/issues/5023
+     * We can safely ignore it in the production environment to avoid hammering Sentry and other
+     * libraries relying on `window.addEventListener('error', callback)`.
+     */
+    const isResizeObserverLoopError = error.message === 'ResizeObserver loop limit exceeded'
+
+    if (process.env.NODE_ENV === 'production' && isResizeObserverLoopError) {
+        error.stopImmediatePropagation()
+        error.stopPropagation()
```
Comment (umpox) [inline client/web/src/sentry.ts:18]:
```suggestion
        error.stopImmediatePropagation()
```

Do you need both here? I thought `stopImmediatePropagation` does the same and more

File: client/web/src/util/dom.ts (hunk 1 above)
Comment (valerybugakov) [reply client/web/src/util/dom.ts:25]: It doesn't throw in the Chrome console and Typescript doesn't complain 🧐

Screenshot 2021-10-05 at 12 19 33

**Review by umpox — APPROVED** [review]

File: client/web/src/sentry.ts (hunk 2 above)
Comment (valerybugakov) [reply client/web/src/sentry.ts:18]: Good catch 👍

**Review by vovakulikov — APPROVED** [review]

**Pull request merged**